app/js/lib/foxbox/services/light.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'Light';
+
+// Private members.
+const p = Object.freeze({
+  // Public getter only properties.
+  manufacturer: Symbol('manufacturer'),
+  model: Symbol('model'),
+  name: Symbol('name'),
+
+  // Private getter only properties.
+  adapter: Symbol('adapter'),
+  getters: Symbol('getters'),
+  setters: Symbol('setters'),
+});
+
+export default class LightService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+
+    // Public getter only properties.
+    this[p.manufacturer] = props.properties.manufacturer;
+    this[p.model] = props.properties.model;
+    this[p.name] = props.properties.name;
+
+    // Private getter only properties.
+    this[p.adapter] = props.adapter;
+    this[p.getters] = props.getters;
+    this[p.setters] = props.setters;
+
+    Object.seal(this);
+  }
+
+  // Public getter only properties.
+  get manufacturer() {
+    return this[p.manufacturer];
+  }
+
+  get model() {
+    return this[p.model];
+  }
+
+  get name() {
+    return this[p.name];
+  }
+
+  get type() {
+    return TYPE;
+  }
+
+  // Public dynamic property getters.
+  isAvailable() {
+    return this[p.query]('available')
+      .then((response) => {
+        for (let id in response) {
+          return response[id].OnOff === 'On';
+        }
+      });
+  }
+
+  isOn() {
+    return this[p.query]('LightOn')
+      .then((response) => {
+        for (let id in response) {
+          return response[id].OnOff === 'On';
+        }
+      });
+  }
+
+  // Public methods.
+  /**
+   * Turn the bulb on or off.
+   *
+   * @param {boolean} on Whether to turn it on (true) or off (false).
+   * @return {Promise}
+   */
+  turn(on) {
+    const value = on ? 'On' : 'Off';
+    const operation = this.getOperation(this[p.setters], 'LightOn');
+
+    return this.performSetOperation(operation, value);
+  }
+
+  /**
+   * Turn on the bulb.
+   *
+   * @return {Promise}
+   */
+  turnOn() {
```

Comment:
Oh the API is similar!
I don't really like device translator. Specifically because we use the term `service` and `translator` doesn't bring any meaning here.
I like `service wrapper` though. Maybe we should use that term to refer to these classes.